gitleaks: add page (#12653)
tldr-pages/tldr: pages/common/gitleaks.md

# gitleaks

> > A scanner for detecting secrets and API keys leaked in Git repositories.

Reviewer: ```suggestion
> Detect secrets and API keys leaked in Git repositories.
```
The previous description was a valid one, but we generally start with a verb in the imperative mood.

Or:

```suggestion
> Scan for secrets and API keys leaked in Git repositories.
```

> > For more information, see: <https://github.com/gitleaks/gitleaks>.

Reviewer 2: ```suggestion
> More information: <https://github.com/gitleaks/gitleaks>.
```
This syntax is hard-coded across our pages, so I would suggest using the same here.

- Scan a remote repository:

`gitleaks detect --repo-url {{https://github.com/username/repository.git}}`

- Scan a local directory:

> `gitleaks detect --source {{path/to/repo}}`

Reviewer 2: ```suggestion
`gitleaks detect --source {{path/to/repository}}`
```
We generally prefer verbose descriptions and placeholders wherever possible.

Reviewer: I think we should make this clearer in the style guide, it currently says:

> #### Naming
> - Use short but descriptive placeholders, such as {{path/to/source_file}} or {{path/to/wallet.txt}}.

We could clarify this improving the example:

> #### Naming
> - Use short but descriptive placeholders, such as {{path/to/source_file}} or {{path/to/wallet.txt}} instead of {{path/to/src}} or {{path/to/file.txt}}.

- Output scan results to a JSON file:

> `gitleaks detect --source {{path/to/repo}} --report {{path/to/report.json}}`

Reviewer: ```suggestion
`gitleaks detect --source {{path/to/repository}} --report {{path/to/report.json}}`
```

- Use a custom rules file:

> `gitleaks detect --source {{path/to/repo}} --config-path {{path/to/config.toml}}`

Reviewer: ```suggestion
`gitleaks detect --source {{path/to/repository}} --config-path {{path/to/config.toml}}`
```

- Start scanning from a specific commit:

> `gitleaks detect --source {{path/to/repo}} --log-opts {{--since=commit_id}}`

Reviewer: ```suggestion
`gitleaks detect --source {{path/to/repository}} --log-opts {{--since=commit_id}}`
```
